File: 2000.go
func reversePrefix(word string, ch byte) string {
    i := strings.Index(word, string(ch));
    if i > -1 {
        s := []rune(word);
        for j := 0; j < i; j, i = j + 1, i - 1 {
            s[i], s[j] = s[j], s[i];
        }
        return string(s);
    }
    return word;
}

File: 2000.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        size_t i = word.find(ch);
        if (i != string::npos) {
            size_t j = 0;
            while (j < i) {
                swap(word[j], word[i]);
                j++, i--;
            }
            return word;
        }
        return word;
    }
};
